### PR TITLE
fix for parent context not setting child flags

### PR DESCRIPTION
**Pull request by anicoll**
## What type of PR is this?

- [ x] bug

## What this PR does / why we need it:

It fixes a simple bug for parent contexts not setting specific flags (`IntSlice`, `Int64Slice`)

## Which issue(s) this PR fixes:

Fixes https://github.com/urfave/cli/issues/1213

## Release Notes

```release-note
* Fixes an issue with flags being set by parent context's (`IntSlice`, `Int64Slice`)
```

### CODE REVIEW

File: docs/v2/manual.md [hunk 1]
```diff
@@ -649,7 +649,7 @@ func main() {
 
   app := &cli.App{
     Action: func(c *cli.Context) error {
-      fmt.Println("yaml ist rad")
+      fmt.Println("--test value.*default: 0")
```
Comment (anicoll) [inline docs/v2/manual.md:652]: apologies for this part. I am not really sure whats happening her I just tried a few things locally to get the tests to pass. Seems like they fail on master rather than any change I have made.

Please let me know if you would like me to revert this change.

**Comment by anicoll** [comment]
Any movement on this one?

**Review by rliebz — APPROVED** [review]

**Comment by lynncyrin** [comment]
@anicoll Can you pull in the default branch to fix the failing test please 🙏🏼

**Comment by lynncyrin** [comment]
Actually, I was incorrect, it looks like codecov is what's failing 👀

**Comment by anicoll** [comment]
@lynncyrin fixed. :)

**Review by lynncyrin — APPROVED** [review]
🚚

**Review by rliebz — APPROVED** [review]
